Update shell configuration, test pnpm setup, simplify (#91)
upleveled/system-setup: windows.md

   ```bash
   choco install slack -y
   ```
7. Close PowerShell and open it again as administrator (like in step 2). Copy the following text and right-click in the blue middle part of the PowerShell window to paste the text. Hit enter.<br><br>
   ```bash
   corepack enable
   corepack prepare pnpm@latest --activate
   Set-ExecutionPolicy -Scope Process -ExecutionPolicy Bypass
   pnpm setup
   ```
   This uses Corepack to install `pnpm`, and configures `pnpm`'s global bin directory.<br><br>

>    Install `@upleveled/preflight`, a program we will use in the course, to verify that the previous commands were successful: copy the following text, paste it in the terminal and hit return.<br><br>

Reviewer: seems like we should have empty lines around the paragraphs and code block, right?

Author: Added empty lines around the codeblock
[d295f33](https://github.com/upleveled/system-setup/pull/91/commits/d295f336fbe786aae30ba1358e97b15c8c00e0df)

   ```bash
   pnpm add --global @upleveled/preflight

>    ```

Reviewer: it looks like this removal of the "Close PowerShell and open it again as administrator ..." may be introducing a new error:

in [this comment](https://github.com/upleveled/system-setup/issues/41#issuecomment-2082303483) it appears that `pnpm setup` does indeed set environment variables on the system and needs to have a new terminal started (see the bottom of the screenshot)

![Screenshot 2024-04-29 104040](https://github.com/upleveled/system-setup/assets/1935696/2d28836a-b252-44d8-a513-52349321aec4)

Author: I successfully uninstalled Node and removed all the Windows environment variables used by pnpm. Reinstalled Node with `choco` and followed the System Setup steps.

In the screenshot, pnpm confirms that the setup is complete with the message 

```bash
Setup complete. Open a new terminal to start using pnpm.
```

![pnpm-install](https://github.com/user-attachments/assets/124dde2c-3dab-46c6-a1b4-f7e0d0dd77f1)
At the bottom of the screenshot, we can see that `pnpm` logs `Setup complete. Open a new terminal to start using pnpm`

I added the `Close PowerShell and open it again as administrator (like in step 2).` to the instructions in this commit
[43408f5](https://github.com/upleveled/system-setup/pull/91/commits/43408f5dcd8891b07e46794de0687a52d20ce962)

>    Verify that your output looks similar to the "Successful" output below. If it looks very different (for example like the "Failing" output), repeat the pnpm installation commands at the start of this step.

Reviewer: let's polish this (in all 3 guides)

```suggestion
   Verify that your output looks similar to the "Successful" output below. If it looks very different (for example like the "Failing" output), repeat the pnpm installation commands at the start of this step.
```

>    Verify that your output looks similar to the "Successful" output below. If it looks very different (for example like the "Failing" output), repeat the pnpm installation commands at the start of this step.

Reviewer: let's polish this (in all 3 guides)

```suggestion
   Verify that your output looks similar to the "Successful" output below. If it looks very different (either like the "Failing" output or some other different output), repeat the pnpm installation commands at the start of this step.
```

>    Verify that your output looks similar to the "Successful" output below. If it looks very different (for example like the "Failing" output), repeat the pnpm installation commands at the start of this step.

Reviewer: let's polish this (in all 3 guides)

```suggestion
   Verify that your output looks similar to the "Successful" output below. You can ignore differences in version numbers and package names - the important part is "Done in ..." on the last line.

   If it looks very different (either like the "Failing" output or like some other different output), repeat the pnpm installation commands at the start of this step.
```

>    Verify that your output looks similar to the "Successful" output below. If it looks very different (for example like the "Failing" output), repeat the pnpm installation commands at the start of this step.

Reviewer: let's polish this (in all 3 guides)

```suggestion
   Verify that your output looks similar to the "Successful" output below. You can ignore any differences in version numbers and package names - the important part is "Done in ..." on the last line.

   If it looks very different (either like the "Failing" output or like some other different output), repeat the pnpm installation commands at the start of this step.
```

   ```bash
   choco install slack -y
   ```

> 7. Close PowerShell and open it again as administrator (like in step 2). Copy the following text and right-click in the blue middle part of the PowerShell window to paste the text. Hit enter.<br><br>

Author: We need to reopen the terminal because, after installing with choco, the environment variables changed, as explained in this [comment](https://github.com/upleveled/system-setup/pull/91#issuecomment-2270597504).

Reviewer: [the linked comment](https://github.com/upleveled/system-setup/pull/91#issuecomment-2270597504) doesn't explain a reason why we should move the "close and reopen"  from step 8 to step 7 in the Windows guide

the comment explains that `node -v` didn't work, but that's not a command that we run in step 7

check into this again - if we actually have a problem with `corepack` being unavailable, then of course we can make this or other changes. but I'm guessing that we don't have an issue here.

Author: I checked it on Windows again, removed `nodejs-lts` from my system with all folders, also `pnpm`, and started new with the installation of `node`. For the installation, I used the same commands as the System Setup Guide as seen in the first screenshot. We can see the last command where I used `corepack enable`, and we get an error message that `corepack` is not recognized. 
![corepack-missing](https://github.com/user-attachments/assets/d026b7d8-209e-4aec-b29c-e92aea2a685e)

After restarting the terminal, `corepack` is recognized and I can use the latest `pnpm` version to install Preflight
![corapack-restart](https://github.com/user-attachments/assets/c039d221-9462-4ebc-8650-684b6cd87c01)

Reviewer: removing the "Close PowerShell and open it again ..." from the following step may introduce a new error:

- https://github.com/upleveled/system-setup/pull/91#discussion_r1726895994

   corepack enable
   corepack prepare pnpm@latest --activate
   pnpm setup

>    source "$USERPROFILE/.bash_profile"

Reviewer: since [we'll switch from `$USERPROFILE` to `~`](https://github.com/upleveled/system-setup/pull/91#issuecomment-2264962880), we'll have a provlem if we keep this here - [it appears PowerShell doesn't support tilde](https://www.reddit.com/r/PowerShell/comments/14wdeih/tilde/)

we could move the Hyper install + configuration with Git Bash up earlier (eg. as a part of step 6 or a new step between step 6 and 7)

downside: the checking of the shell settings are further apart, and could become inconsistent.

Reviewer: @ProchaLu decided to remove the `source` command from step 7 of the Windows guide in the latest commits

   This installs some VS Code extensions we will need.<br><br>

11. We recommend installing and using Chrome so that you have the same DevTools as others.<br><br>
    If you don't have Chrome installed yet, you can install it with Chocolatey. To do this, copy the following text and right-click in the blue middle part of the PowerShell window to paste the text. Hit enter.<br><br>
    ```bash
    choco install googlechrome -y
    ```
    This uses Chocolatey to install Chrome.<br><br>
12. Install the following Chrome Extensions:
    - [React Developer tools Chrome Extension](https://chrome.google.com/webstore/detail/react-developer-tools/fmkadmapgofadopljbjfkapdkoienihi?hl=en)
    - [Refined GitHub Chrome Extension](https://chrome.google.com/webstore/detail/refined-github/hlepfoohegkhhmjieoechaddaejaokhf?hl=en)
    - [Socket Security Chrome Extension](https://chrome.google.com/webstore/detail/socket-security/jbcobpbfgkhmjfpjjepkcocalmpkiaop?hl=en)
    - [Web Vitals Chrome Extension](https://chrome.google.com/webstore/detail/web-vitals/ahfhijdlegdabablpippeagghigmibma?hl=en)

> 13. Close PowerShell and open it again as administrator (like in step 2)<br><br>

Author: This step is not necessary anymore. Removed and fixed the numbers in this commit [5324859](https://github.com/upleveled/system-setup/pull/91/commits/5324859e37acf59ceeb60c89e9b90795d80d96f0)

    - [Refined GitHub Chrome Extension](https://chrome.google.com/webstore/detail/refined-github/hlepfoohegkhhmjieoechaddaejaokhf?hl=en)
    - [Socket Security Chrome Extension](https://chrome.google.com/webstore/detail/socket-security/jbcobpbfgkhmjfpjjepkcocalmpkiaop?hl=en)
    - [Web Vitals Chrome Extension](https://chrome.google.com/webstore/detail/web-vitals/ahfhijdlegdabablpippeagghigmibma?hl=en)

> 12. Close PowerShell and open it again as administrator (like in step 2)<br><br>

Author: Removed because it is unnecessary to restart PowerShell at this step

    Now let's set an environment variable to tell PostgreSQL where to find the programs and where to put the data. Copy and run each of these lines separately in Hyper:

    ```bash
    echo "export PATH=\$PATH:\"/c/Program Files/PostgreSQL/16/bin\"" >> ~/.bash_profile
    echo "export PGDATA=\"/c/Program Files/PostgreSQL/16/data\"" >> ~/.bash_profile
    echo "export PSQL_PAGER=\"less --chop-long-lines --header 1\"" >> ~/.bash_profile

>     source ~/.bash_profile

Reviewer: maybe let's split this out to a separate PR, to avoid this getting lost in this PR

Author: Created a new PR for these changes 
https://github.com/upleveled/system-setup/pull/94

    Click on the small `x` at the top right of the virtual device frame to stop the virtual device - this will save a snapshot to make starting the virtual device faster in the future.

24. If you don't have one yet, create a Google account [here](https://accounts.google.com/signup?hl=en). Make a note of the email address associated with this account for usage in later steps.
25. If you don't have one yet, create a GitHub account [here](https://github.com/join). Make sure to set a name.

    If you already have a GitHub account and you haven't set a name on GitHub yet, go to the [GitHub Profile Settings](https://github.com/settings/profile) and add a name:<br><br>
    <img src="./general-github-profile-settings.png"><br><br>
    We will use this name in the next step.<br><br>

> 26. For this step, we'll need to **edit some of the information in the commands** by adding our own information.<br><br>

Reviewer: looks like the numbering is off, starting with 28 (outside of the diff, below)

Author: fixed numbers [ecb57fe](https://github.com/upleveled/system-setup/pull/91/commits/ecb57fe2e42067fd1de662f9002f6fcaf212d11c)
